src/evm/protocol/uniswap_v3/state.rs

```diff
@@ -248,12 +249,16 @@ impl ProtocolSim for UniswapV3State {
 
     fn get_amount_out(
         &self,
-        amount_in: U256,
+        amount_in: BigUint,
         token_a: &ERC20Token,
         token_b: &ERC20Token,
     ) -> Result<GetAmountOutResult, SimulationError> {
         let zero_for_one = token_a < token_b;
-        let amount_specified = I256::checked_from_sign_and_abs(Sign::Positive, amount_in).unwrap();
+        let amount_specified = I256::checked_from_sign_and_abs(
+            Sign::Positive,
+            U256::from_be_slice(&amount_in.to_bytes_be()),
```

Comment:
Unfortunately this allocates a Vec<u8> and immediately references it... this is not great but I couldn't find a better way to do it...